ykcompile/src/lib.rs

```diff
@@ -1511,8 +1551,21 @@ impl<TT> TraceCompiler<TT> {
     }
 
     /// Finish compilation and return the executable code that was assembled.
-    fn finish(self) -> dynasmrt::ExecutableBuffer {
-        self.asm.finalize().unwrap()
+    fn finish(self, debug: bool) -> dynasmrt::ExecutableBuffer {
+        let buf = self.asm.finalize().unwrap();
+        if debug {
+            // In debug mode the memory section which contains the compiled trace is marked as
+            // writeable, which enables gdb/lldb to set breakpoints within the compiled code.
```

Comment:
How??? I have never had problems with setting breakpoints for cranelift-simplejit, which only enables `PROT_EXEC`.

Reply:
Indeed. How? I've yet to find an answer to that question. If you have an idea I'd be very curious to hear it. See the stackoverflow discussion for more details.